controller/src/main/java/io/pravega/controller/server/ControllerService.java

```diff
@@ -344,49 +344,79 @@ public static long nextRequestId() {
+                                                                   final long createTimestamp, long requestId) {
+        Timer timer = new Timer();
         return Futures.exceptionallyExpecting(streamStore.getState(scope, stream, true, null, executor),
-                e -> Exceptions.unwrap(e) instanceof StoreException.DataNotFoundException, State.UNKNOWN)
-                      .thenCompose(state -> {
-                          if (state.equals(State.UNKNOWN) || state.equals(State.CREATING)) {
-                              return streamMetadataTasks.createStreamRetryOnLockFailure(scope,
-                                      stream,
-                                      streamConfig,
-                                      createTimestamp, 10, 
-                                      requestId).thenApplyAsync(status -> {
-                                  reportCreateStreamMetrics(scope, stream, streamConfig.getScalingPolicy().getMinNumSegments(), 
-                                          status, timer.getElapsed());
-                                  return CreateStreamStatus.newBuilder().setStatus(status).build();
-                              }, executor);
-                          } else {
-                              log.info(requestId, "Stream {} already exists ", NameUtils.getScopedStreamName(scope, stream));
-                              return CompletableFuture.completedFuture(
-                                      CreateStreamStatus.newBuilder().setStatus(CreateStreamStatus.Status.STREAM_EXISTS).build());
-                          }
-                      });
+                        e -> Exceptions.unwrap(e) instanceof StoreException.DataNotFoundException, State.UNKNOWN)
+                .thenCompose(state -> {
+                    if (state.equals(State.UNKNOWN) || state.equals(State.CREATING)) {
+                        return streamMetadataTasks.createStreamRetryOnLockFailure(scope,
+                                stream,
+                                streamConfig,
+                                createTimestamp, 10,
+                                requestId).thenApplyAsync(status -> {
+                            reportCreateStreamMetrics(scope, stream, streamConfig.getScalingPolicy().getMinNumSegments(),
+                                    status, timer.getElapsed());
+                            return CreateStreamStatus.newBuilder().setStatus(status).build();
+                        }, executor);
+                    } else {
+                        log.info(requestId, "Stream {} already exists ", NameUtils.getScopedStreamName(scope, stream));
+                        return CompletableFuture.completedFuture(
+                                CreateStreamStatus.newBuilder().setStatus(CreateStreamStatus.Status.STREAM_EXISTS).build());
+                    }
+                });
+    }
+
+    private void validate(final StreamConfiguration streamConfig, final long createTimestamp) {
```

Comment:
should we name this function as validateStreamConfig instead? Are there other places that could make use of this method?

Reply:
this method is validating timestamp too